cygwin build failed for pyinstaller 6.10

applicable to 6.8-6.10 and develop branch:

      [19/21] Compiling src/main.c
      ../../src/pyi_main.c: In function ‘pyi_main’:
      ../../src/pyi_main.c:410:9: error: ‘ret’ undeclared (first use in this function)
        410 |         ret = cygwin_conv_path(CCP_POSIX_TO_WIN_W | CCP_RELATIVE, pyi_ctx->application_home_dir, dllpath_w, PYI_PATH_MAX);
            |         ^~~
      ../../src/pyi_main.c:410:9: note: each undeclared identifier is reported only once for each function it appears in

      Waf: Leaving directory `/tmp/pip-install-fjdd4f0p/pyinstaller_280306b2082a4db59eb280efd05c2f18/bootloader/build/debug'
      Build failed
       -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)

syntax error: need to declare "int ret" before usage in the function

--- a/bootloader/src/pyi_main.c
+++ b/bootloader/src/pyi_main.c
@@ -419,8 +419,7 @@ pyi_main(struct PYI_CONTEXT *pyi_ctx)
 
 #if defined(__CYGWIN__)
         /* Cygwin */
-        ret = cygwin_conv_path(CCP_POSIX_TO_WIN_W | CCP_RELATIVE, pyi_ctx->application_home_dir, dllpath_w, PYI_PATH_MAX);
-        if (ret != 0) {
+        if (cygwin_conv_path(CCP_POSIX_TO_WIN_W | CCP_RELATIVE, pyi_ctx->application_home_dir, dllpath_w, PYI_PATH_MAX) != 0) {
             PYI_PERROR("cygwin_conv_path", "Failed to convert DLL search path!\n");
             return -1;
         }

--- a/news/8814.bootloader.rst
+++ b/news/8814.bootloader.rst
@@ -1,0 +1,2 @@
+(Cygwin) Fix missing-variable-error when compiling bootloader under
+Cygwin (regression introduced in PyInstaller v6.8).

